doc/stp/SONiC_PVST_HLD.md

```diff
@@ -0,0 +1,736 @@
+Ethernet13  128  4    Y    N        FORWARDING 0           8000002438eefbc3 8000002438eefbc3 
+```
+
+- show spanning_tree bpdu_guard
+This command displays the interfaces which are BPDU guard enabled and also the state if the interface is disabled due to BPDU guard.
+```
+show spanning_tree bpdu_guard
+PortNum            Shutdown      Port shut
+                   Configured    due to BPDU guard
+-------------------------------------------------
+Ethernet1            Yes          Yes
+Ethernet2            Yes          No
+Port-Channel2        No           NA
+```
+
+-show spanning_tree root_guard
+This command displays the interfaces where root guard is active and the pending time for root guard timer expiry
+```
+Root guard timeout: 120 secs
+
+Port         VLAN    Current State
+-------------------------------------------------
+Ethernet1    1       Inconsistent state (102 seconds left on timer)
+Ethernet8    100     Consistent state
+```
+
+- show spanning_tree statistics 
+- show spanning_tree statistics vlan <vlanid\> 
+This command displays the spanning-tree bpdu statistics. Statistics will be synced to APP DB every 10 seconds.
+```
+VLAN 100 - STP instance 3
+--------------------------------------------------------------------
+PortNum           BPDU Tx     BPDU Rx     TCN Tx      TCN Rx             
+Ethernet13        10	      4           3          4
+PortChannel15     20	      6           4          1
+```
+
+
+### 3.6.4 Debug Commands
+Following debug commands will be supported for enabling additional logging which can be viewed in /var/log/syslog and packet logs can be viewed in /var/log/stp_dbg.log(in STP container)
+- debug spanning_tree config_bpdu
+- debug spanning_tree tcn_bpdu
+- debug spanning_tree event
+- debug spanning_tree interface <ifname\>
+- debug spanning_tree verbose
+- debug spanning_tree vlan <id\>
+
+To disable the debugging controls enabled '-d' or '--disable' option can be used, an example of disabling config_bpdu debugging is shown below -
+- debug spanning_tree config_bpdu -d
+
+Follow commands can be used to reset and display the debugging controls enabled respectively
+- debug spanning_tree reset
+- debug spanning_tree show
+
+Following debug commands will be supported for displaying internal data structures
+- debug spanning_tree dump global
+- debug spanning_tree dump vlan <vid\>
+- debug spanning_tree dump interface <vid\> <ifname\>
+
+### 3.6.5 Clear Commands
+Following clear commands will be supported 
+- sonic-clear spanning_tree statistics
+- sonic-clear spanning_tree statistics vlan <vid\>
+- sonic-clear spanning_tree statistics vlan-interface <vid\> <ifname\>
+- sonic-clear spanning_tree statistics interface <ifname\>
+
+### 3.6.6 REST API Support
+REST APIs is not supported in this release
+
+# 4 Flow Diagrams
+
+## 4.1 Global spanning-tree mode
+
+![STP](images/Global_PVST_enable.png "Figure 2: Global PVST mode enable")
+
+__Figure 2: Global PVST mode enable__
+
```

Comment:
There might be config_db entries that have added VLANs to STP instances but this would not have taken effect until Global PVST is enable.  So when GloblaPVST is enabled, this flow should be extended to add VLANs to the STP instances.

Reply:
Just to clarify, global pvst must be enabled for any other spanning tree config to be accepted. In this case (add VLANs to STP instance flow) is pretty much same as described in "Figure 4 STP VLAN enable".